src/btreemap/proptests.rs

```diff
@@ -196,5 +200,33 @@ fn execute_operation<M: Memory>(
                 assert_eq!(btree.remove(&k), Some(v));
             }
         }
+        Operation::PopLast => {
+            assert_eq!(std_btree.len(), btree.len() as usize);
+            if std_btree.is_empty() {
+                return;
+            }
+
+            let idx = std_btree.len();
+
+            if let Some(entry) = btree.iter().skip(idx).take(1).next() {
```

Comment:
I don't understand why we need this `if` statement. Is it not sufficient to compare `std_btree.pop_last()` with `btree.pop_last()` and check that they're equal?

Reply:
get_max and remove call value reading twice, which is unnecessary. Especially when there are N items of data, this overhead will be * N